packages/apidom-ns-json-schema-draft-4/src/refractor/visitors/json-schema/index.ts

```diff
@@ -1,33 +1,95 @@
 import { Mixin } from 'ts-mixer';
-import { always } from 'ramda';
+import { always, defaultTo } from 'ramda';
+import { isNonEmptyString, isUndefined } from 'ramda-adjunct';
+import {
+  ObjectElement,
+  ArrayElement,
+  isStringElement,
+  cloneDeep,
+  toValue,
+} from '@swagger-api/apidom-core';
 
 import FixedFieldsVisitor, {
   FixedFieldsVisitorOptions,
   SpecPath,
 } from '../generics/FixedFieldsVisitor.ts';
+import ParentSchemaAwareVisitor, {
+  ParentSchemaAwareVisitorOptions,
+} from './ParentSchemaAwareVisitor.ts';
 import FallbackVisitor, { FallbackVisitorOptions } from '../FallbackVisitor.ts';
 import JSONSchemaElement from '../../../elements/JSONSchema.ts';
+import { isJSONSchemaElement } from '../../../predicates.ts';
 
 /**
  * @public
  */
 export interface JSONSchemaVisitorOptions
   extends FixedFieldsVisitorOptions,
+    ParentSchemaAwareVisitorOptions,
     FallbackVisitorOptions {}
 
 /**
  * @public
  */
-class JSONSchemaVisitor extends Mixin(FixedFieldsVisitor, FallbackVisitor) {
+class JSONSchemaVisitor extends Mixin(
+  FixedFieldsVisitor,
+  ParentSchemaAwareVisitor,
+  FallbackVisitor,
+) {
   declare public element: JSONSchemaElement;
 
   declare protected readonly specPath: SpecPath<['document', 'objects', 'JSONSchema']>;
 
+  protected readonly default$schema = 'http://json-schema.org/draft-04/schema#';
+
   constructor(options: JSONSchemaVisitorOptions) {
     super(options);
     this.element = new JSONSchemaElement();
     this.specPath = always(['document', 'objects', 'JSONSchema']);
   }
+
+  ObjectElement(objectElement: ObjectElement) {
+    this.handle$schema(objectElement);
+    this.handleId(objectElement);
+
+    // for further processing consider this Schema Element as parent for all embedded Schema Elements
+    this.parent = this.element;
+
+    return FixedFieldsVisitor.prototype.ObjectElement.call(this, objectElement);
+  }
+
+  handle$schema(objectElement: ObjectElement): void {
```

Comment:
`handle$schema` - in Draft 4/5 we define and use as public method.

Then in Draft 6 we want to use JSONSchemaVisitor.prototype.handle$schema.call from Draft 4/5.
Then in Draft 7 we want to use JSONSchemaVisitor.prototype.handle$schema.call from Draft 6.

The point is to just always rely only on previous version of JSON Schema namespace instead of all namespaces relying on Draft 4.